internal/util/http.go

```diff
@@ -25,6 +25,17 @@ import (
 	"github.com/coreos/ignition/internal/version"
 )
 
+// A timeout of 0 indicates no timeout
+var (
+	httpResponseHeaderTimeout int
```

Comment:
AFAIK we don't reuse them at all and instead create a new one every time we make a request (which is not great). We could switch to something like GetHttpClient, which would hand you either a new one (if not created yet) or one that has already been created, but we would still need a global to store the HttpClient.